hawk-java/src/main/java/so/hawk/java/catcher/HawkHttpUtils.java

```diff
@@ -0,0 +1,120 @@
+package so.hawk.java.catcher;
+
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+public class HawkHttpUtils {
+
+    /**
+     * Decodes the integration ID from a Base64-encoded token.
+     *
+     * @param token the encoded token
+     * @return the integration ID
+     */
+    public static String decodeToken(String token) {
+      try {
+        String decodedJson = new String(Base64.getDecoder().decode(token));
+        JSONObject json = new JSONObject(decodedJson);
+        return json.getString("integrationId");
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Invalid token: Unable to decode Base64 JSON.");
+      }
+    }
+
+    /**
+     * Sends an error message to the server.
+     *
+     * @param hawkInstance the Hawk instance to use
+     * @param e            the exception to log
+     */
+    public static void sendError(Hawk hawkInstance, Exception e) {
```

Comment:
HTTP Utils should not know about Hawk format and "Error" term. It should be responsible only for http transport

Reply:
composing Hawk data format should be somewhere in other place